app/models/spree/stock/availability_validator_decorator.rb

```diff
@@ -24,18 +24,26 @@ def quantity_in_shipment(line_item)
 
   def line_item_shipment(line_item)
     return line_item.target_shipment if line_item.target_shipment
-    return line_item.order.shipments.first if line_item.order.present? && line_item.order.shipments.any?
+    return line_item.order.shipments.first if line_item.order.andand.shipments.any?
   end
 
-  # This is the spree v2.0.4 implementation of validate
-  # But using the calculated quantity instead of the line_item.quantity.
+  # Overrides Spree v2.0.4 validate method version to:
+  #   - scope variants to hub and thus acivate variant overrides
+  #   - use calculated quantity instead of the line_item.quantity
+  #   - rely on Variant.can_supply? instead of Stock::Quantified.can_supply?
+  #       so that it works correctly for variant overrides
   def validate_quantity(line_item, quantity)
-    quantifier = Spree::Stock::Quantifier.new(line_item.variant_id)
-    return if quantifier.can_supply? quantity
+    line_item.scoper.scope(line_item.variant)
```

Comment:
Instead of exposing the scoper, you could offer `line_item.scope_variant`.

I still have a concern about the performance of the scoper used in this way, but that can probably be solved later on.